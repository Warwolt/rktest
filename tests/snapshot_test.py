import os
import subprocess

TEST_EXECUTABLE = './build/Debug/tests' if os.name == 'nt' else './build/tests'
FAILING_TEST_EXECUTABLE = './build/Debug/failing_tests' if os.name == 'nt' else './build/failing_tests'


def run_test_exe(exe, args: [str] = []) -> str:
    result = subprocess.run([exe, '--rktest_print_time=0', '--rktest_print_filenames=0',
                            '--rktest_color=no'] + args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True)
    print('cmd:', ' '.join(result.args))
    return result.stdout


def test_no_args(snapshot):
    actual = run_test_exe(TEST_EXECUTABLE)
    assert actual == snapshot


def test_wildcard_match(snapshot):
    actual = run_test_exe(TEST_EXECUTABLE, ['--rktest_filter=*'])
    assert actual == snapshot


def test_suffix_match(snapshot):
    actual = run_test_exe(TEST_EXECUTABLE, ['--rktest_filter=*equal'])
    assert actual == snapshot


def test_prefix_match(snapshot):
    actual = run_test_exe(TEST_EXECUTABLE, ['--rktest_filter=integer*'])
    assert actual == snapshot


def test_infix_match(snapshot):
    actual = run_test_exe(TEST_EXECUTABLE, ['--rktest_filter=*tests*'])
    assert actual == snapshot


def test_failing_tests(snapshot):
    actual = run_test_exe(FAILING_TEST_EXECUTABLE)
    assert actual == snapshot


def test_print_help(snapshot):
    actual = run_test_exe(TEST_EXECUTABLE, ['--help'])
    assert actual == snapshot


def test_pass_bad_arg(snapshot):
    actual = run_test_exe(TEST_EXECUTABLE, ['--badargument'])
    assert actual == snapshot
