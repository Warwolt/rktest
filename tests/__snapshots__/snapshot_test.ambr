# serializer version: 1
# name: test_failing_tests
  '''
  [==========] Running 37 tests from 5 test suites.
  [----------] Global test environment set-up.
  [----------] 1 tests from disabled_tests
  [ DISABLED ] disabled_tests.DISABLED_this_test_should_not_run
  [ RUN      ] disabled_tests.this_test_should_run 
  [       OK ] disabled_tests.this_test_should_run 
  [----------] 1 tests from disabled_tests 
  
  [----------] 4 tests from float_tests
  [ RUN      ] float_tests.float_equal 
  error: Expected equality of these values:
    float_sum
      Which is: 1.00000000
    0.9f
      Which is: 0.89999998
   
  [  FAILED  ] float_tests.float_equal 
  [ RUN      ] float_tests.float_equal_info 
  error: Expected equality of these values:
    float_sum
      Which is: 1.00000000
    0.9f
      Which is: 0.89999998
  Hello world!
  
  [  FAILED  ] float_tests.float_equal_info 
  [ RUN      ] float_tests.double_equal 
  error: Expected equality of these values:
    double_sum
      Which is: 1.00000000
    0.9
      Which is: 0.90000000
   
  [  FAILED  ] float_tests.double_equal 
  [ RUN      ] float_tests.double_equal_info 
  error: Expected equality of these values:
    double_sum
      Which is: 1.00000000
    0.9
      Which is: 0.90000000
  Hello world!
  
  [  FAILED  ] float_tests.double_equal_info 
  [----------] 4 tests from float_tests 
  
  [----------] 16 tests from integer_tests
  [ RUN      ] integer_tests.expect_true 
  error: Value of: `int_sum1 == 3`:
    Actual: false
  Expected: true
   
  [  FAILED  ] integer_tests.expect_true 
  [ RUN      ] integer_tests.expect_true_info 
  error: Value of: `int_sum2 == 7`:
    Actual: false
  Expected: true
  int_sum2 = 17
  
  [  FAILED  ] integer_tests.expect_true_info 
  [ RUN      ] integer_tests.expect_false 
  error: Value of: `int_sum1 == 1 + 2 + 50`:
    Actual: true
  Expected: false
   
  [  FAILED  ] integer_tests.expect_false 
  [ RUN      ] integer_tests.expect_false_info 
  [       OK ] integer_tests.expect_false_info 
  [ RUN      ] integer_tests.expect_equal 
  error: Expected equality of these values:
    int_sum1
      Which is: 53
    3
   
  [  FAILED  ] integer_tests.expect_equal 
  [ RUN      ] integer_tests.expect_equal_info 
  error: Expected equality of these values:
    int_sum2
      Which is: 17
    7
  int_sum2 = 17
  
  [  FAILED  ] integer_tests.expect_equal_info 
  [ RUN      ] integer_tests.expect_not_equal 
  [       OK ] integer_tests.expect_not_equal 
  [ RUN      ] integer_tests.expect_not_equal_info 
  [       OK ] integer_tests.expect_not_equal_info 
  [ RUN      ] integer_tests.expect_less_than 
  error: Expected (int_sum1) < (int_sum2), actual: 53 vs 17
   
  [  FAILED  ] integer_tests.expect_less_than 
  [ RUN      ] integer_tests.expect_less_than_info 
  error: Expected (int_sum1) < (int_sum2), actual: 53 vs 17
  int_sum2 = 17
  
  [  FAILED  ] integer_tests.expect_less_than_info 
  [ RUN      ] integer_tests.expect_less_than_equal 
  error: Expected (int_sum1) <= (int_sum2), actual: 53 vs 17
   
  [  FAILED  ] integer_tests.expect_less_than_equal 
  [ RUN      ] integer_tests.expect_less_than_equal_info 
  error: Expected (int_sum1) <= (int_sum2), actual: 53 vs 17
  int_sum2 = 17
  
  [  FAILED  ] integer_tests.expect_less_than_equal_info 
  [ RUN      ] integer_tests.expect_greater_than 
  error: Expected (int_sum2) > (int_sum1), actual: 17 vs 53
   
  [  FAILED  ] integer_tests.expect_greater_than 
  [ RUN      ] integer_tests.expect_greater_than_info 
  error: Expected (int_sum2) > (int_sum1), actual: 17 vs 53
  int_sum2 = 17
  
  [  FAILED  ] integer_tests.expect_greater_than_info 
  [ RUN      ] integer_tests.expect_greater_than_equal 
  error: Expected (int_sum2) >= (int_sum1), actual: 17 vs 53
   
  [  FAILED  ] integer_tests.expect_greater_than_equal 
  [ RUN      ] integer_tests.expect_greater_than_equal_info 
  error: Expected (int_sum2) >= (int_sum1), actual: 17 vs 53
  int_sum2 = 17
  
  [  FAILED  ] integer_tests.expect_greater_than_equal_info 
  [----------] 16 tests from integer_tests 
  
  [----------] 8 tests from string_tests
  [ RUN      ] string_tests.strings_equal 
  error: Expected equality of these values:
    str1
      Which is: elloh
    "hello"
   
  [  FAILED  ] string_tests.strings_equal 
  [ RUN      ] string_tests.strings_equal_info 
  error: Expected equality of these values:
    str2
      Which is: dlorw
    "world"
  str2 = dlorw
  
  [  FAILED  ] string_tests.strings_equal_info 
  [ RUN      ] string_tests.strings_not_equal 
  [       OK ] string_tests.strings_not_equal 
  [ RUN      ] string_tests.strings_not_equal_info 
  [       OK ] string_tests.strings_not_equal_info 
  [ RUN      ] string_tests.strings_case_equal 
  error: Expected equality of these values:
    str1
      Which is: elloh
    "Hello"
  Ignoring case
   
  [  FAILED  ] string_tests.strings_case_equal 
  [ RUN      ] string_tests.strings_case_equal_info 
  error: Expected equality of these values:
    str2
      Which is: dlorw
    "World"
  Ignoring case
  str2 = dlorw
  
  [  FAILED  ] string_tests.strings_case_equal_info 
  [ RUN      ] string_tests.strings_case_not_equal 
  [       OK ] string_tests.strings_case_not_equal 
  [ RUN      ] string_tests.strings_case_not_equal_info 
  [       OK ] string_tests.strings_case_not_equal_info 
  [----------] 8 tests from string_tests 
  
  [----------] 8 tests from wildcard_match_tests
  [ RUN      ] wildcard_match_tests.empty_pattern_matches_only_empty_string 
  [       OK ] wildcard_match_tests.empty_pattern_matches_only_empty_string 
  [ RUN      ] wildcard_match_tests.literal_pattern_matches_only_exact_literal 
  [       OK ] wildcard_match_tests.literal_pattern_matches_only_exact_literal 
  [ RUN      ] wildcard_match_tests.single_asterisk_matches_any_string 
  [       OK ] wildcard_match_tests.single_asterisk_matches_any_string 
  [ RUN      ] wildcard_match_tests.literal_then_asterisk_does_prefix_match 
  [       OK ] wildcard_match_tests.literal_then_asterisk_does_prefix_match 
  [ RUN      ] wildcard_match_tests.asterisk_then_literal_does_suffix_match 
  [       OK ] wildcard_match_tests.asterisk_then_literal_does_suffix_match 
  [ RUN      ] wildcard_match_tests.prefix_and_suffix_match 
  [       OK ] wildcard_match_tests.prefix_and_suffix_match 
  [ RUN      ] wildcard_match_tests.infix_match 
  [       OK ] wildcard_match_tests.infix_match 
  [ RUN      ] wildcard_match_tests.double_asterisk 
  [       OK ] wildcard_match_tests.double_asterisk 
  [----------] 8 tests from wildcard_match_tests 
  
  [----------] Global test environment tear-down.
  [==========] 37 tests from 5 test suites ran. 
  [  PASSED  ] 16 tests.
  [  FAILED  ] 21 tests, listed below:
  [  FAILED  ] float_tests.float_equal
  [  FAILED  ] float_tests.float_equal_info
  [  FAILED  ] float_tests.double_equal
  [  FAILED  ] float_tests.double_equal_info
  [  FAILED  ] integer_tests.expect_true
  [  FAILED  ] integer_tests.expect_true_info
  [  FAILED  ] integer_tests.expect_false
  [  FAILED  ] integer_tests.expect_equal
  [  FAILED  ] integer_tests.expect_equal_info
  [  FAILED  ] integer_tests.expect_less_than
  [  FAILED  ] integer_tests.expect_less_than_info
  [  FAILED  ] integer_tests.expect_less_than_equal
  [  FAILED  ] integer_tests.expect_less_than_equal_info
  [  FAILED  ] integer_tests.expect_greater_than
  [  FAILED  ] integer_tests.expect_greater_than_info
  [  FAILED  ] integer_tests.expect_greater_than_equal
  [  FAILED  ] integer_tests.expect_greater_than_equal_info
  [  FAILED  ] string_tests.strings_equal
  [  FAILED  ] string_tests.strings_equal_info
  [  FAILED  ] string_tests.strings_case_equal
  [  FAILED  ] string_tests.strings_case_equal_info
  
   21 FAILED TESTS
    YOU HAVE 3 DISABLED TESTS
  
  '''
# ---
# name: test_infix_match
  '''
  Note: Test filter = *tests*
  [==========] Running 37 tests from 5 test suites.
  [----------] Global test environment set-up.
  [----------] 1 tests from disabled_tests
  [ DISABLED ] disabled_tests.DISABLED_this_test_should_not_run
  [ RUN      ] disabled_tests.this_test_should_run 
  [       OK ] disabled_tests.this_test_should_run 
  [----------] 1 tests from disabled_tests 
  
  [----------] 4 tests from float_tests
  [ RUN      ] float_tests.float_equal 
  [       OK ] float_tests.float_equal 
  [ RUN      ] float_tests.float_equal_info 
  [       OK ] float_tests.float_equal_info 
  [ RUN      ] float_tests.double_equal 
  [       OK ] float_tests.double_equal 
  [ RUN      ] float_tests.double_equal_info 
  [       OK ] float_tests.double_equal_info 
  [----------] 4 tests from float_tests 
  
  [----------] 16 tests from integer_tests
  [ RUN      ] integer_tests.expect_true 
  [       OK ] integer_tests.expect_true 
  [ RUN      ] integer_tests.expect_true_info 
  [       OK ] integer_tests.expect_true_info 
  [ RUN      ] integer_tests.expect_false 
  [       OK ] integer_tests.expect_false 
  [ RUN      ] integer_tests.expect_false_info 
  [       OK ] integer_tests.expect_false_info 
  [ RUN      ] integer_tests.expect_equal 
  [       OK ] integer_tests.expect_equal 
  [ RUN      ] integer_tests.expect_equal_info 
  [       OK ] integer_tests.expect_equal_info 
  [ RUN      ] integer_tests.expect_not_equal 
  [       OK ] integer_tests.expect_not_equal 
  [ RUN      ] integer_tests.expect_not_equal_info 
  [       OK ] integer_tests.expect_not_equal_info 
  [ RUN      ] integer_tests.expect_less_than 
  [       OK ] integer_tests.expect_less_than 
  [ RUN      ] integer_tests.expect_less_than_info 
  [       OK ] integer_tests.expect_less_than_info 
  [ RUN      ] integer_tests.expect_less_than_equal 
  [       OK ] integer_tests.expect_less_than_equal 
  [ RUN      ] integer_tests.expect_less_than_equal_info 
  [       OK ] integer_tests.expect_less_than_equal_info 
  [ RUN      ] integer_tests.expect_greater_than 
  [       OK ] integer_tests.expect_greater_than 
  [ RUN      ] integer_tests.expect_greater_than_info 
  [       OK ] integer_tests.expect_greater_than_info 
  [ RUN      ] integer_tests.expect_greater_than_equal 
  [       OK ] integer_tests.expect_greater_than_equal 
  [ RUN      ] integer_tests.expect_greater_than_equal_info 
  [       OK ] integer_tests.expect_greater_than_equal_info 
  [----------] 16 tests from integer_tests 
  
  [----------] 8 tests from string_tests
  [ RUN      ] string_tests.strings_equal 
  [       OK ] string_tests.strings_equal 
  [ RUN      ] string_tests.strings_equal_info 
  [       OK ] string_tests.strings_equal_info 
  [ RUN      ] string_tests.strings_not_equal 
  [       OK ] string_tests.strings_not_equal 
  [ RUN      ] string_tests.strings_not_equal_info 
  [       OK ] string_tests.strings_not_equal_info 
  [ RUN      ] string_tests.strings_case_equal 
  [       OK ] string_tests.strings_case_equal 
  [ RUN      ] string_tests.strings_case_equal_info 
  [       OK ] string_tests.strings_case_equal_info 
  [ RUN      ] string_tests.strings_case_not_equal 
  [       OK ] string_tests.strings_case_not_equal 
  [ RUN      ] string_tests.strings_case_not_equal_info 
  [       OK ] string_tests.strings_case_not_equal_info 
  [----------] 8 tests from string_tests 
  
  [----------] 8 tests from wildcard_match_tests
  [ RUN      ] wildcard_match_tests.empty_pattern_matches_only_empty_string 
  [       OK ] wildcard_match_tests.empty_pattern_matches_only_empty_string 
  [ RUN      ] wildcard_match_tests.literal_pattern_matches_only_exact_literal 
  [       OK ] wildcard_match_tests.literal_pattern_matches_only_exact_literal 
  [ RUN      ] wildcard_match_tests.single_asterisk_matches_any_string 
  [       OK ] wildcard_match_tests.single_asterisk_matches_any_string 
  [ RUN      ] wildcard_match_tests.literal_then_asterisk_does_prefix_match 
  [       OK ] wildcard_match_tests.literal_then_asterisk_does_prefix_match 
  [ RUN      ] wildcard_match_tests.asterisk_then_literal_does_suffix_match 
  [       OK ] wildcard_match_tests.asterisk_then_literal_does_suffix_match 
  [ RUN      ] wildcard_match_tests.prefix_and_suffix_match 
  [       OK ] wildcard_match_tests.prefix_and_suffix_match 
  [ RUN      ] wildcard_match_tests.infix_match 
  [       OK ] wildcard_match_tests.infix_match 
  [ RUN      ] wildcard_match_tests.double_asterisk 
  [       OK ] wildcard_match_tests.double_asterisk 
  [----------] 8 tests from wildcard_match_tests 
  
  [----------] Global test environment tear-down.
  [==========] 37 tests from 5 test suites ran. 
  [  PASSED  ] 37 tests.
  
    YOU HAVE 3 DISABLED TESTS
  
  '''
# ---
# name: test_no_args
  '''
  [==========] Running 37 tests from 5 test suites.
  [----------] Global test environment set-up.
  [----------] 1 tests from disabled_tests
  [ DISABLED ] disabled_tests.DISABLED_this_test_should_not_run
  [ RUN      ] disabled_tests.this_test_should_run 
  [       OK ] disabled_tests.this_test_should_run 
  [----------] 1 tests from disabled_tests 
  
  [----------] 4 tests from float_tests
  [ RUN      ] float_tests.float_equal 
  [       OK ] float_tests.float_equal 
  [ RUN      ] float_tests.float_equal_info 
  [       OK ] float_tests.float_equal_info 
  [ RUN      ] float_tests.double_equal 
  [       OK ] float_tests.double_equal 
  [ RUN      ] float_tests.double_equal_info 
  [       OK ] float_tests.double_equal_info 
  [----------] 4 tests from float_tests 
  
  [----------] 16 tests from integer_tests
  [ RUN      ] integer_tests.expect_true 
  [       OK ] integer_tests.expect_true 
  [ RUN      ] integer_tests.expect_true_info 
  [       OK ] integer_tests.expect_true_info 
  [ RUN      ] integer_tests.expect_false 
  [       OK ] integer_tests.expect_false 
  [ RUN      ] integer_tests.expect_false_info 
  [       OK ] integer_tests.expect_false_info 
  [ RUN      ] integer_tests.expect_equal 
  [       OK ] integer_tests.expect_equal 
  [ RUN      ] integer_tests.expect_equal_info 
  [       OK ] integer_tests.expect_equal_info 
  [ RUN      ] integer_tests.expect_not_equal 
  [       OK ] integer_tests.expect_not_equal 
  [ RUN      ] integer_tests.expect_not_equal_info 
  [       OK ] integer_tests.expect_not_equal_info 
  [ RUN      ] integer_tests.expect_less_than 
  [       OK ] integer_tests.expect_less_than 
  [ RUN      ] integer_tests.expect_less_than_info 
  [       OK ] integer_tests.expect_less_than_info 
  [ RUN      ] integer_tests.expect_less_than_equal 
  [       OK ] integer_tests.expect_less_than_equal 
  [ RUN      ] integer_tests.expect_less_than_equal_info 
  [       OK ] integer_tests.expect_less_than_equal_info 
  [ RUN      ] integer_tests.expect_greater_than 
  [       OK ] integer_tests.expect_greater_than 
  [ RUN      ] integer_tests.expect_greater_than_info 
  [       OK ] integer_tests.expect_greater_than_info 
  [ RUN      ] integer_tests.expect_greater_than_equal 
  [       OK ] integer_tests.expect_greater_than_equal 
  [ RUN      ] integer_tests.expect_greater_than_equal_info 
  [       OK ] integer_tests.expect_greater_than_equal_info 
  [----------] 16 tests from integer_tests 
  
  [----------] 8 tests from string_tests
  [ RUN      ] string_tests.strings_equal 
  [       OK ] string_tests.strings_equal 
  [ RUN      ] string_tests.strings_equal_info 
  [       OK ] string_tests.strings_equal_info 
  [ RUN      ] string_tests.strings_not_equal 
  [       OK ] string_tests.strings_not_equal 
  [ RUN      ] string_tests.strings_not_equal_info 
  [       OK ] string_tests.strings_not_equal_info 
  [ RUN      ] string_tests.strings_case_equal 
  [       OK ] string_tests.strings_case_equal 
  [ RUN      ] string_tests.strings_case_equal_info 
  [       OK ] string_tests.strings_case_equal_info 
  [ RUN      ] string_tests.strings_case_not_equal 
  [       OK ] string_tests.strings_case_not_equal 
  [ RUN      ] string_tests.strings_case_not_equal_info 
  [       OK ] string_tests.strings_case_not_equal_info 
  [----------] 8 tests from string_tests 
  
  [----------] 8 tests from wildcard_match_tests
  [ RUN      ] wildcard_match_tests.empty_pattern_matches_only_empty_string 
  [       OK ] wildcard_match_tests.empty_pattern_matches_only_empty_string 
  [ RUN      ] wildcard_match_tests.literal_pattern_matches_only_exact_literal 
  [       OK ] wildcard_match_tests.literal_pattern_matches_only_exact_literal 
  [ RUN      ] wildcard_match_tests.single_asterisk_matches_any_string 
  [       OK ] wildcard_match_tests.single_asterisk_matches_any_string 
  [ RUN      ] wildcard_match_tests.literal_then_asterisk_does_prefix_match 
  [       OK ] wildcard_match_tests.literal_then_asterisk_does_prefix_match 
  [ RUN      ] wildcard_match_tests.asterisk_then_literal_does_suffix_match 
  [       OK ] wildcard_match_tests.asterisk_then_literal_does_suffix_match 
  [ RUN      ] wildcard_match_tests.prefix_and_suffix_match 
  [       OK ] wildcard_match_tests.prefix_and_suffix_match 
  [ RUN      ] wildcard_match_tests.infix_match 
  [       OK ] wildcard_match_tests.infix_match 
  [ RUN      ] wildcard_match_tests.double_asterisk 
  [       OK ] wildcard_match_tests.double_asterisk 
  [----------] 8 tests from wildcard_match_tests 
  
  [----------] Global test environment tear-down.
  [==========] 37 tests from 5 test suites ran. 
  [  PASSED  ] 37 tests.
  
    YOU HAVE 3 DISABLED TESTS
  
  '''
# ---
# name: test_pass_bad_arg
  '''
  Error: Unrecognized argument --badargument
  
  This program is a unit test runner built using RK Test.
  
  Usage:
  
    --rktest_color=(yes|no|auto)
      Enable/disable colored output. The default is auto.
  
    --rktest_filter=PATTERN
      Run only the tests that matches the globbing pattern. * matches against
      any number of characters, and ? matches any single character.
  
    --rktest_print_time=0
      Disable printing out the elapsed time for test cases and test suites.
  
    --rktest_print_filenames=0
      Disable printing out the filename of a test case on assert failure.
  
  '''
# ---
# name: test_prefix_match
  '''
  Note: Test filter = integer*
  [==========] Running 16 tests from 1 test suites.
  [----------] Global test environment set-up.
  [----------] 16 tests from integer_tests
  [ RUN      ] integer_tests.expect_true 
  [       OK ] integer_tests.expect_true 
  [ RUN      ] integer_tests.expect_true_info 
  [       OK ] integer_tests.expect_true_info 
  [ RUN      ] integer_tests.expect_false 
  [       OK ] integer_tests.expect_false 
  [ RUN      ] integer_tests.expect_false_info 
  [       OK ] integer_tests.expect_false_info 
  [ RUN      ] integer_tests.expect_equal 
  [       OK ] integer_tests.expect_equal 
  [ RUN      ] integer_tests.expect_equal_info 
  [       OK ] integer_tests.expect_equal_info 
  [ RUN      ] integer_tests.expect_not_equal 
  [       OK ] integer_tests.expect_not_equal 
  [ RUN      ] integer_tests.expect_not_equal_info 
  [       OK ] integer_tests.expect_not_equal_info 
  [ RUN      ] integer_tests.expect_less_than 
  [       OK ] integer_tests.expect_less_than 
  [ RUN      ] integer_tests.expect_less_than_info 
  [       OK ] integer_tests.expect_less_than_info 
  [ RUN      ] integer_tests.expect_less_than_equal 
  [       OK ] integer_tests.expect_less_than_equal 
  [ RUN      ] integer_tests.expect_less_than_equal_info 
  [       OK ] integer_tests.expect_less_than_equal_info 
  [ RUN      ] integer_tests.expect_greater_than 
  [       OK ] integer_tests.expect_greater_than 
  [ RUN      ] integer_tests.expect_greater_than_info 
  [       OK ] integer_tests.expect_greater_than_info 
  [ RUN      ] integer_tests.expect_greater_than_equal 
  [       OK ] integer_tests.expect_greater_than_equal 
  [ RUN      ] integer_tests.expect_greater_than_equal_info 
  [       OK ] integer_tests.expect_greater_than_equal_info 
  [----------] 16 tests from integer_tests 
  
  [----------] Global test environment tear-down.
  [==========] 16 tests from 1 test suites ran. 
  [  PASSED  ] 16 tests.
  
  '''
# ---
# name: test_print_help
  '''
  
  This program is a unit test runner built using RK Test.
  
  Usage:
  
    --rktest_color=(yes|no|auto)
      Enable/disable colored output. The default is auto.
  
    --rktest_filter=PATTERN
      Run only the tests that matches the globbing pattern. * matches against
      any number of characters, and ? matches any single character.
  
    --rktest_print_time=0
      Disable printing out the elapsed time for test cases and test suites.
  
    --rktest_print_filenames=0
      Disable printing out the filename of a test case on assert failure.
  
  '''
# ---
# name: test_suffix_match
  '''
  Note: Test filter = *equal
  [==========] Running 10 tests from 3 test suites.
  [----------] Global test environment set-up.
  [----------] 2 tests from float_tests
  [ RUN      ] float_tests.float_equal 
  [       OK ] float_tests.float_equal 
  [ RUN      ] float_tests.double_equal 
  [       OK ] float_tests.double_equal 
  [----------] 2 tests from float_tests 
  
  [----------] 4 tests from integer_tests
  [ RUN      ] integer_tests.expect_equal 
  [       OK ] integer_tests.expect_equal 
  [ RUN      ] integer_tests.expect_not_equal 
  [       OK ] integer_tests.expect_not_equal 
  [ RUN      ] integer_tests.expect_less_than_equal 
  [       OK ] integer_tests.expect_less_than_equal 
  [ RUN      ] integer_tests.expect_greater_than_equal 
  [       OK ] integer_tests.expect_greater_than_equal 
  [----------] 4 tests from integer_tests 
  
  [----------] 4 tests from string_tests
  [ RUN      ] string_tests.strings_equal 
  [       OK ] string_tests.strings_equal 
  [ RUN      ] string_tests.strings_not_equal 
  [       OK ] string_tests.strings_not_equal 
  [ RUN      ] string_tests.strings_case_equal 
  [       OK ] string_tests.strings_case_equal 
  [ RUN      ] string_tests.strings_case_not_equal 
  [       OK ] string_tests.strings_case_not_equal 
  [----------] 4 tests from string_tests 
  
  [----------] Global test environment tear-down.
  [==========] 10 tests from 3 test suites ran. 
  [  PASSED  ] 10 tests.
  
  '''
# ---
# name: test_wildcard_match
  '''
  Note: Test filter = *
  [==========] Running 37 tests from 5 test suites.
  [----------] Global test environment set-up.
  [----------] 1 tests from disabled_tests
  [ DISABLED ] disabled_tests.DISABLED_this_test_should_not_run
  [ RUN      ] disabled_tests.this_test_should_run 
  [       OK ] disabled_tests.this_test_should_run 
  [----------] 1 tests from disabled_tests 
  
  [----------] 4 tests from float_tests
  [ RUN      ] float_tests.float_equal 
  [       OK ] float_tests.float_equal 
  [ RUN      ] float_tests.float_equal_info 
  [       OK ] float_tests.float_equal_info 
  [ RUN      ] float_tests.double_equal 
  [       OK ] float_tests.double_equal 
  [ RUN      ] float_tests.double_equal_info 
  [       OK ] float_tests.double_equal_info 
  [----------] 4 tests from float_tests 
  
  [----------] 16 tests from integer_tests
  [ RUN      ] integer_tests.expect_true 
  [       OK ] integer_tests.expect_true 
  [ RUN      ] integer_tests.expect_true_info 
  [       OK ] integer_tests.expect_true_info 
  [ RUN      ] integer_tests.expect_false 
  [       OK ] integer_tests.expect_false 
  [ RUN      ] integer_tests.expect_false_info 
  [       OK ] integer_tests.expect_false_info 
  [ RUN      ] integer_tests.expect_equal 
  [       OK ] integer_tests.expect_equal 
  [ RUN      ] integer_tests.expect_equal_info 
  [       OK ] integer_tests.expect_equal_info 
  [ RUN      ] integer_tests.expect_not_equal 
  [       OK ] integer_tests.expect_not_equal 
  [ RUN      ] integer_tests.expect_not_equal_info 
  [       OK ] integer_tests.expect_not_equal_info 
  [ RUN      ] integer_tests.expect_less_than 
  [       OK ] integer_tests.expect_less_than 
  [ RUN      ] integer_tests.expect_less_than_info 
  [       OK ] integer_tests.expect_less_than_info 
  [ RUN      ] integer_tests.expect_less_than_equal 
  [       OK ] integer_tests.expect_less_than_equal 
  [ RUN      ] integer_tests.expect_less_than_equal_info 
  [       OK ] integer_tests.expect_less_than_equal_info 
  [ RUN      ] integer_tests.expect_greater_than 
  [       OK ] integer_tests.expect_greater_than 
  [ RUN      ] integer_tests.expect_greater_than_info 
  [       OK ] integer_tests.expect_greater_than_info 
  [ RUN      ] integer_tests.expect_greater_than_equal 
  [       OK ] integer_tests.expect_greater_than_equal 
  [ RUN      ] integer_tests.expect_greater_than_equal_info 
  [       OK ] integer_tests.expect_greater_than_equal_info 
  [----------] 16 tests from integer_tests 
  
  [----------] 8 tests from string_tests
  [ RUN      ] string_tests.strings_equal 
  [       OK ] string_tests.strings_equal 
  [ RUN      ] string_tests.strings_equal_info 
  [       OK ] string_tests.strings_equal_info 
  [ RUN      ] string_tests.strings_not_equal 
  [       OK ] string_tests.strings_not_equal 
  [ RUN      ] string_tests.strings_not_equal_info 
  [       OK ] string_tests.strings_not_equal_info 
  [ RUN      ] string_tests.strings_case_equal 
  [       OK ] string_tests.strings_case_equal 
  [ RUN      ] string_tests.strings_case_equal_info 
  [       OK ] string_tests.strings_case_equal_info 
  [ RUN      ] string_tests.strings_case_not_equal 
  [       OK ] string_tests.strings_case_not_equal 
  [ RUN      ] string_tests.strings_case_not_equal_info 
  [       OK ] string_tests.strings_case_not_equal_info 
  [----------] 8 tests from string_tests 
  
  [----------] 8 tests from wildcard_match_tests
  [ RUN      ] wildcard_match_tests.empty_pattern_matches_only_empty_string 
  [       OK ] wildcard_match_tests.empty_pattern_matches_only_empty_string 
  [ RUN      ] wildcard_match_tests.literal_pattern_matches_only_exact_literal 
  [       OK ] wildcard_match_tests.literal_pattern_matches_only_exact_literal 
  [ RUN      ] wildcard_match_tests.single_asterisk_matches_any_string 
  [       OK ] wildcard_match_tests.single_asterisk_matches_any_string 
  [ RUN      ] wildcard_match_tests.literal_then_asterisk_does_prefix_match 
  [       OK ] wildcard_match_tests.literal_then_asterisk_does_prefix_match 
  [ RUN      ] wildcard_match_tests.asterisk_then_literal_does_suffix_match 
  [       OK ] wildcard_match_tests.asterisk_then_literal_does_suffix_match 
  [ RUN      ] wildcard_match_tests.prefix_and_suffix_match 
  [       OK ] wildcard_match_tests.prefix_and_suffix_match 
  [ RUN      ] wildcard_match_tests.infix_match 
  [       OK ] wildcard_match_tests.infix_match 
  [ RUN      ] wildcard_match_tests.double_asterisk 
  [       OK ] wildcard_match_tests.double_asterisk 
  [----------] 8 tests from wildcard_match_tests 
  
  [----------] Global test environment tear-down.
  [==========] 37 tests from 5 test suites ran. 
  [  PASSED  ] 37 tests.
  
    YOU HAVE 3 DISABLED TESTS
  
  '''
# ---
